filters/model.js

```diff
@@ -0,0 +1,314 @@
+const filter = module.exports;
+
+const { 
+  JavaGenerator,
+  TypeHelpers,
+  ModelKind,
+  FormatHelpers,
+  CommonModel,
+
+  JAVA_CONSTRAINTS_PRESET,
+  JAVA_JACKSON_PRESET,
+  JAVA_DESCRIPTION_PRESET,
+} = require('@asyncapi/generator-model-sdk');
+
+let javaGenerator = undefined;
+
+function getCombinedSchema({ model, inputModel, type = 'oneOf' }) {
+  let combined = undefined;
+
+  const originalSchema = model.originalSchema;
+  if (originalSchema && (originalSchema[type])) {
+    combined = originalSchema[type];
+  }
+
+  const ref = model.$ref;
+  const refModel = ref && inputModel && inputModel.models[ref];
+  if (refModel) {
+    const originalSchema = refModel.originalSchema;
+    if (originalSchema && (originalSchema[type])) {
+      combined = originalSchema[type];
+    }
+  }
+
+  return combined;
+}
+
+function getTypeFromCombinedSchema({ combinedSchema, type = 'OneOf' }) {
+  for (const schema of combinedSchema) {
+    // render as Object type
+    if (schema.type !== "object") {
+      type = "Object";
+      break;
+    }
+    type += FormatHelpers.toPascalCase(schema['x-parser-schema-id']);
+  }
+  return type;
+}
+
+const INLINE_ENUM_PRESET = {
+  class: {
+    async property({ renderer, propertyName, property, content }) {
+      if (ModelKind.ENUM !== TypeHelpers.extractKind(property)) {
+        return content;
+      }
+      const enumName = FormatHelpers.toPascalCase(`${propertyName}Enum`);
+    
+      propertyName = FormatHelpers.toCamelCase(propertyName);
+      const renderedProperty = `private ${enumName} ${propertyName};`;
+    
+      const newSchema = Object.assign({}, property.originalSchema);
+      newSchema.$id = enumName;
+      const enumModel = await javaGenerator.render(newSchema);
+    
+      const annotation = renderer.renderAnnotation('Valid');
+      return renderer.renderBlock([enumModel, annotation, renderedProperty]);
+    },
+    getter({ renderer, propertyName, property, content }) {
+      if (ModelKind.ENUM !== TypeHelpers.extractKind(property)) {
+        return content;
+      }
+
+      propertyName = FormatHelpers.toCamelCase(propertyName);
+      const getterName = FormatHelpers.toPascalCase(propertyName);
+      const type = FormatHelpers.toPascalCase(`${propertyName}Enum`);
+      return `public ${type} get${getterName}() { return this.${propertyName}; }`;
+    },
+    setter({ renderer, propertyName, property, content }) {
+      if (ModelKind.ENUM !== TypeHelpers.extractKind(property)) {
+        return content;
+      }
+
+      propertyName = FormatHelpers.toCamelCase(propertyName);
+      const setterName = FormatHelpers.toPascalCase(propertyName);
+      const type = FormatHelpers.toPascalCase(`${propertyName}Enum`);
+      return `public void set${setterName}(${type} ${propertyName}) { this.${propertyName} = ${propertyName}; }`;
+    },
+  }
+}
+
+const INLINE_ONE_ANY_OF = {
```

Comment:
Is it like this maybe?

```suggestion
const INLINE_ANY_OF = {
```